Library/Homebrew/formula.rb

```diff
@@ -553,13 +562,33 @@ def print_tap_action options={}
 
   # True if this formula is provided by Homebrew itself
   def core_formula?
-    path == Formula.path(name)
+    path == Formula.core_path(name)
   end
 
-  def self.path name
+  def self.core_path name
     Pathname.new("#{HOMEBREW_LIBRARY}/Formula/#{name.downcase}.rb")
   end
```

Comment:
Let's do it the other way round; make the new method have the new functionality.